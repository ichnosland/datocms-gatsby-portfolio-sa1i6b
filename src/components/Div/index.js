/**
*
* Div
*
*/

import styled, { css } from 'styled-components';
import media from 'style/mediainjector';

const Div = styled.div`
  display: ${(props) => props.display ? props.display : 'block'};
  position: ${(props) => props.position ? props.position : 'relative'};
  width: ${(props) => props.width};
  max-width: ${(props) => props.maxWidth ? props.maxWidth : '100%'};
  height: ${(props) => props.height};
  color: ${(props) => props.color ? props.color : 'inherit'};
  text-align: ${(props) => props.align};
  float: ${(props) => props.float ? props.float : 'none'};
  margin: ${(props) => props.margin ? props.margin : '0'};
  padding: ${(props) => props.padding ? props.padding : '0'};
  border: ${(props) => props.border ? props.border : 'none'};
  border-width: ${(props) => props.borderwidth};
  border-style: ${(props) => props.borderstyle};
  border-color: ${(props) => props.bordercolor};
  background-color: ${(props) => props.bgColor ? props.bgColor : 'transparent'};
  border-radius: ${(props) => props.radius};
  overflow: ${(props) => props.overflow ? props.overflow : 'visible'};
  user-select: ${(props) => props.userSelect};
  pointer-events: ${(props) => props.pointerEvents};
  z-index: ${(props) => props.zindex};
  ${(props) => props.nobreak && css`
    page-break-inside: avoid;
  `}
  ${media.print`
    ${({ noPrint }) => noPrint && css`
        display:none;
    `}
  `}
`;

export default Div;
