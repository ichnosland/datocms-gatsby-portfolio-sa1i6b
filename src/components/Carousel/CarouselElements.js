/**
*
* CarouselElements
*
*/

import styled, { css } from 'styled-components';

import Div from 'components/Div';
import { NoStyleButton } from 'components/Button';

export const CarouselWrap = styled(Div)`
  width:  ${(props) => {
    if (props.buttonWidth) return `calc(100% - (${props.buttonWidth} * 2))`;
    return '100%';
  }};
  margin: 0 auto;
  overflow: hidden;
`;

export const CarouselContainer = styled(Div)`
  display: flex;
  margin: 0 auto;
  transition: ${(props) => props.sliding ? 'none' : 'transform 0.5s ease'};
  transform: ${({ sliding, noTransforms, direction }) => {
    if (!noTransforms && !sliding) return 'translateX(calc(-100%))';
    if (!noTransforms && direction === 'prev') return 'translateX(calc(2 * (-100%)))';
    if (!noTransforms) return 'translateX(0%)';

    return '';
  }};
`;

export const CarouselSlide = styled(Div)`
  flex: 1 0 100%;
  width: 100%;
  order: ${(props) => props.order};
`;

export const ArrowBox = styled.button`
  ${NoStyleButton}
  display: flex;
  flex-direction: column;
  position: absolute;
  top: 0;
  width: ${(props) => props.width ? props.width : '40px'};
  height: ${(props) => props.height ? props.height : '100%'};
  justify-content: center;
  align-items: center;
  background-color: ${(props) => props.bgcolor};
  ${({ sx }) => sx && css`
    left: 0;
  `}
  ${({ dx }) => dx && css`
    right: 0;
  `}
  ${({ hidden }) => hidden && css`
    display: none;
  `}
`;
