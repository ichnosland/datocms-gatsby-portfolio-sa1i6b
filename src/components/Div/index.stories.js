import React from 'react';
import { storiesOf } from '@storybook/react';
import { withNotes } from '@storybook/addon-notes';
import { withKnobs, text, boolean, color } from '@storybook/addon-knobs';
import Div from './index';


const nota = 'La props nobreak assegna \'page-break-inside: avoid;\' \n quella noPrint \'display:none;\' in stampa';

storiesOf('Div', module)
  .addDecorator(withKnobs)
  .addWithJSX('Div', withNotes(nota)(() => (
    <Div
      display={text('display', 'block')}
      position={text('position', 'relative')}
      width={text('width', 'auto')}
      maxWidth={text('maxWidth', '100%')}
      height={text('height', '')}
      color={color('color', 'inherit')}
      align={text('align (tex-align)', '')}
      float={text('float', 'none')}
      margin={text('margin', '0')}
      padding={text('padding', '0')}
      radius={text('radius', '')}
      bgColor={color('bgColor', 'transparent')}
      border={text('border', '')}
      borderwidth={text('borderwidth', '')}
      borderstyle={text('borderstyle', '')}
      bordercolor={color('bordercolor', '')}
      overflow={text('overflow', 'visible')}
      userSelect={text('userSelect', 'auto')}
      pointerEvents={text('pointerEvents', 'auto')}
      zindex={text('zindex', 'auto')}
      nobreak={boolean('nobreak', false)}
      noPrint={boolean('noPrint', true)}
    >
      <p>Ciao, sono all&apos;interno di un Div. Usa le props per personalizzarlo :)</p>
    </Div>
  )));
