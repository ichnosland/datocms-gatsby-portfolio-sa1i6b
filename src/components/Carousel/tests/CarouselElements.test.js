import React from 'react';
import { shallow } from 'enzyme';

import Div from '../Div';
import { CarouselWrap, CarouselSlide, CarouselContainer, ArrowBox } from '../CarouselElements';

describe('<CarouselWrap />', () => {
  it('should render a Div', () => {
    const renderedComponent = shallow(
      <CarouselWrap />
    );
    expect(renderedComponent.find(Div).length).toBe(1);
  });

  it('should calc its width 100% minus double the props buttonWidth', () => {
    const buttonWidth = '10px';
    const renderedComponent = shallow(
      <CarouselWrap buttonWidth={buttonWidth} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<CarouselSlide />', () => {
  it('should render a Div', () => {
    const renderedComponent = shallow(
      <CarouselSlide />
    );
    expect(renderedComponent.find(Div).length).toBe(1);
  });
});

describe('<CarouselContainer />', () => {
  it('should render a Div', () => {
    const renderedComponent = shallow(
      <CarouselContainer />
    );
    expect(renderedComponent.find(Div).length).toBe(1);
  });

  it('should render its css w/ props flexGrow', () => {
    const flexGrow = '0';
    const renderedComponent = shallow(
      <CarouselContainer flexGrow={flexGrow} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its childrens css w/ props next', () => {
    const renderedComponent = shallow(
      <CarouselContainer next>
        <table>
          <tr><td>table</td></tr>
        </table>
      </CarouselContainer>
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its childrens css w/ props prev', () => {
    const renderedComponent = shallow(
      <CarouselContainer prev>
        <div>
          <p>Ciao</p>
        </div>
      </CarouselContainer>
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should it should animate onwards if props sliding is true', () => {
    const renderedComponent = shallow(
      <CarouselContainer sliding>
        <div>
          <p>Ciao</p>
        </div>
      </CarouselContainer>
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should it should animate backwards if props props sliding is true and direction === "prev"', () => {
    const direction = 'prev';
    const renderedComponent = shallow(
      <CarouselContainer sliding direction={direction}>
        <div>
          <p>Ciao</p>
        </div>
      </CarouselContainer>
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should do not animate if noTransform === true', () => {
    const renderedComponent = shallow(
      <CarouselContainer noTransforms>
        <div>
          <p>Ciao</p>
        </div>
      </CarouselContainer>
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<ArrowBox />', () => {
  it('should render a FlexBox', () => {
    const renderedComponent = shallow(
      <ArrowBox />
    );
    expect(renderedComponent.find('button').length).toBe(1);
  });

  it('should render its css w/ props sx', () => {
    const renderedComponent = shallow(
      <ArrowBox sx />
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props dx', () => {
    const renderedComponent = shallow(
      <ArrowBox dx />
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props width', () => {
    const width = '80px';
    const renderedComponent = shallow(
      <ArrowBox width={width} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props height', () => {
    const height = '300px';
    const renderedComponent = shallow(
      <ArrowBox height={height} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props height', () => {
    const renderedComponent = shallow(
      <ArrowBox hidden />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
