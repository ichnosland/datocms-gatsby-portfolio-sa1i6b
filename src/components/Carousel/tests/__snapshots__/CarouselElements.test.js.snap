// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArrowBox /> should render its css w/ props dx 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  top: 0;
  width: 40px;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  right: 0;
}

.c0:active,
.c0:focus {
  outline: none;
}

<button
  className="c0"
  dx={true}
/>
`;

exports[`<ArrowBox /> should render its css w/ props height 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  top: 0;
  width: 40px;
  height: 300px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:active,
.c0:focus {
  outline: none;
}

<button
  className="c0"
  height="300px"
/>
`;

exports[`<ArrowBox /> should render its css w/ props height 2`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  top: 0;
  width: 40px;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: none;
}

.c0:active,
.c0:focus {
  outline: none;
}

<button
  className="c0"
  hidden={true}
/>
`;

exports[`<ArrowBox /> should render its css w/ props sx 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  top: 0;
  width: 40px;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  left: 0;
}

.c0:active,
.c0:focus {
  outline: none;
}

<button
  className="c0"
/>
`;

exports[`<ArrowBox /> should render its css w/ props width 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  top: 0;
  width: 80px;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:active,
.c0:focus {
  outline: none;
}

<button
  className="c0"
  width="80px"
/>
`;

exports[`<CarouselContainer /> should do not animate if noTransform === true 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 auto;
  -webkit-transition: -webkit-transform 0.5s ease;
  -webkit-transition: transform 0.5s ease;
  transition: transform 0.5s ease;
}

<Div
  className="c0"
  noTransforms={true}
>
  <div>
    <p>
      Ciao
    </p>
  </div>
</Div>
`;

exports[`<CarouselContainer /> should it should animate backwards if props props sliding is true and direction === "prev" 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 auto;
  -webkit-transition: none;
  transition: none;
  -webkit-transform: translateX(calc(2 * (-100%)));
  -ms-transform: translateX(calc(2 * (-100%)));
  transform: translateX(calc(2 * (-100%)));
}

<Div
  className="c0"
  direction="prev"
  sliding={true}
>
  <div>
    <p>
      Ciao
    </p>
  </div>
</Div>
`;

exports[`<CarouselContainer /> should it should animate onwards if props sliding is true 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 auto;
  -webkit-transition: none;
  transition: none;
  -webkit-transform: translateX(0%);
  -ms-transform: translateX(0%);
  transform: translateX(0%);
}

<Div
  className="c0"
  sliding={true}
>
  <div>
    <p>
      Ciao
    </p>
  </div>
</Div>
`;

exports[`<CarouselContainer /> should render its childrens css w/ props next 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 auto;
  -webkit-transition: -webkit-transform 0.5s ease;
  -webkit-transition: transform 0.5s ease;
  transition: transform 0.5s ease;
  -webkit-transform: translateX(calc(-100%));
  -ms-transform: translateX(calc(-100%));
  transform: translateX(calc(-100%));
}

<Div
  className="c0"
  next={true}
>
  <table>
    <tr>
      <td>
        table
      </td>
    </tr>
  </table>
</Div>
`;

exports[`<CarouselContainer /> should render its childrens css w/ props prev 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 auto;
  -webkit-transition: -webkit-transform 0.5s ease;
  -webkit-transition: transform 0.5s ease;
  transition: transform 0.5s ease;
  -webkit-transform: translateX(calc(-100%));
  -ms-transform: translateX(calc(-100%));
  transform: translateX(calc(-100%));
}

<Div
  className="c0"
  prev={true}
>
  <div>
    <p>
      Ciao
    </p>
  </div>
</Div>
`;

exports[`<CarouselContainer /> should render its css w/ props flexGrow 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 auto;
  -webkit-transition: -webkit-transform 0.5s ease;
  -webkit-transition: transform 0.5s ease;
  transition: transform 0.5s ease;
  -webkit-transform: translateX(calc(-100%));
  -ms-transform: translateX(calc(-100%));
  transform: translateX(calc(-100%));
}

<Div
  className="c0"
  flexGrow="0"
/>
`;

exports[`<CarouselWrap /> should calc its width 100% minus double the props buttonWidth 1`] = `
.c0 {
  width: calc(100% - (10px * 2));
  margin: 0 auto;
  overflow: hidden;
}

<Div
  buttonWidth="10px"
  className="c0"
/>
`;
